skimage/feature/tests/test_canny.py

```diff
@@ -105,6 +105,11 @@ def test_invalid_use_quantiles(self):
         self.assertRaises(ValueError, F.canny, image, use_quantiles=True,
                           low_threshold=0.5, high_threshold=-100)
 
+        # Example from issue #4282
+        image = data.camera()
+        self.assertRaises(ValueError, F.canny, image, use_quantiles=True,
```

Comment:
Can you explain why this is not catched by lines 266-271 in _canny.py? Can you justify why these lines (266-271) are still needed? Although I can reproduce that this test fails with master, just by reading the code, I do not understand it well

Reply:
What I understand is that the lines 163 and 168 must be done after the checks and that the checks came too late.

Reply:
Is this correct? Then, you can clean up the function.

Reply:
Yes that's it ;-)

Reply:
I don't get what you mean by "clean up the function."

Reply:
Great. As soon as you remove the unnecessary checks, I approve. ;)

Reply:
I think that there is a misunderstanding here, which unnecessary checks?

Reply:
I would remove this ```
        if high_threshold > 1.0 or low_threshold > 1.0:
            raise ValueError("Quantile thresholds must not be > 1.0")
            raise ValueError("Quantile thresholds must not be > 1.0")
        if high_threshold < 0.0 or low_threshold < 0.0:
        if high_threshold < 0.0 or low_threshold < 0.0:
            raise ValueError("Quantile thresholds must not be < 0.0")
            raise ValueError("Quantile thresholds must not be < 0.0")
```
and handle everything at the top of the function.

Reply:
To me, it's not easy to read to have checks fairly late in a function. The sooner we detect something wrong, the better it is. And as far as I understand, there is no need to delay. High and low thresholds are not modified after the line 168.

Reply:
The problem is that previously the  check for quantiles came after the division of the input low and high thresholds by `dtype_max`. That's why the error was not raised, and that's what is reported in #4437 .

Reply:
>  The sooner we detect something wrong, the better it is
I agree, but I don't see how it can be checked sooner since low and high thresholds can be `None`...

Reply:
Am I wrong if I do this? (I didn't check) https://gist.github.com/sciunto/079fd7d466ca50bd8f6e133f882b035d

Reply:
When `use_quantile` is True, the low and high threshold are not supposed to be normalized, they are simply required to be between 0 and 1.
When it is False, the thresholds are divided by `dtype_max` because the input image is internally cast to float using `img_as_float`, and it is not required to be between 0 and 1, so the second check that you introduced is wrong...